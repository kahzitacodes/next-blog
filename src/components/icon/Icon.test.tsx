import { render, screen } from '@testing-library/react'
import { Icon } from '.'

describe('<Icon />', () => {
  it('should render the correct icon', () => {
    const { container } = render(<Icon name="Anchor" size={32} />)

    const svgElement = screen.getByTestId('Anchor')
    expect(svgElement).toBeInTheDocument()
    expect(svgElement).toHaveAttribute('width', '32')

    expect(container.firstChild).toMatchSnapshot()
  })
})
