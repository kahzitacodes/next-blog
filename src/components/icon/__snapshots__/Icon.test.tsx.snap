// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render the correct icon 1`] = `
<svg
  class="lucide lucide-anchor"
  data-testid="Anchor"
  fill="none"
  height="32"
  stroke="currentColor"
  stroke-linecap="round"
  stroke-linejoin="round"
  stroke-width="2"
  viewBox="0 0 24 24"
  width="32"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M12 22V8"
  />
  <path
    d="M5 12H2a10 10 0 0 0 20 0h-3"
  />
  <circle
    cx="12"
    cy="5"
    r="3"
  />
</svg>
`;
